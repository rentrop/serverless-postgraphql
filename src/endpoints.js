
var endpoint = 'http://localhost:5000/graphql';

if(process.env.USE_AWS_ENDPOINT) {
  endpoint = process.env.AWS_ENDPOINT;
};

module.exports = {
    endpoint: endpoint
};
