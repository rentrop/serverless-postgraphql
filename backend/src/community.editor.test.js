import HttpTransport from 'lokka-transport-http';
import Lokka from 'lokka';
import { endpoint } from './endpoints';

const anonLokka = new Lokka({transport: new HttpTransport(endpoint)});
const communityEditorEmail = 'editor@community.floods';
const communityEditorPassword = 'texasfloods';

async function getToken(email, password) {
  const response = await anonLokka.send(`
    mutation($email:String!, $password:String!) {
      authenticate(input: {email: $email, password: $password}) {
        jwtToken
      }
    }
  `,
  {
    email: email,
    password: password
  });

  return response.authenticate.jwtToken;
}

describe('As a community editor', async () => {
  var lokka;

  beforeAll(async (done) => {
    getToken(communityEditorEmail, communityEditorPassword).then((token) => {
      const headers = {
        'Authorization': 'Bearer '+ token
      };
      lokka = new Lokka({transport: new HttpTransport(endpoint, {headers})});
      done();
    });
  });

  it('should fail to register a new user', async () => {
    try {
      const response = await anonLokka.send(`
        mutation {
          registerUser(input: {
            firstName: "New",
            lastName: "User",
            jobTitle: "Community Admin",
            communityId: 1,
            phoneNumber: "555-6666",
            email: "new@us.er",
            password:"texasfloods",
            role:"floods_community_admin"
          }) {
            user {
              id
            }
          }
        }
    `);
    } catch(e) {
      expect(e).toMatchSnapshot();
    }
  });
});
