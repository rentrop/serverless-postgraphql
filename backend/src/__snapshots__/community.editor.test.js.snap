// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`As a community editor should fail to register a new user 1`] = `[Error: GraphQL Error: permission denied for function register_user]`;
